docs/versioned_docs/version-zenith/guides/620942-github-google-cloud.mdx

```diff
@@ -0,0 +1,329 @@
+
+## Step 1: Create a Google Cloud service account
+
+The Dagger pipeline demonstrated in this tutorial (re)builds a container image of an application every time a new commit is added to the application's repository. It then publishes the container image to Google Artifact Registry and deploys it at a public URL using Google Cloud infrastructure.
+
+This requires the following:
+
+- A Google Cloud service account with all necessary privileges
+- A Google Cloud Run service with a public URL and defined resource/capacity/access rules
+- A Google Artifact Registry repository
+- Access to various Google Cloud APIs
+
+:::info
+This step discusses how to create a Google Cloud service account. If you already have a Google Cloud service account and key for your project, you may skip this step.
+:::
+
+<PartialGoogleCloudServiceAcountKeySetup />
+
+## Step 2: Configure Google Cloud APIs
+
+:::info
+This step discusses how to enable various Google Cloud APIs. If you already have these APIs enabled, you may skip this step.
+:::
+
+<PartialGoogleCloudApiSetup />
+
+## Step 3: Configure a Google Cloud Run service
+
+:::info
+This step discusses how to create a Google Cloud Run service. If you already have a Google Cloud Run service for your project, you may skip this step.
+:::
+
+<PartialGoogleCloudRunSetup />
+
+## Step 4: Configure a Google Artifact Registry repository
+
+:::info
+This step discusses how to create a Google Artifact Registry repository. If you already have a Google Google Artifact Registry repository for your project, you may skip this step.
+:::
+
+<PartialGoogleArtifactRegistrySetup />
+
+## Step 5: Create the Dagger Module
+
+The next step is to create a Dagger Module to do the heavy lifting: build a container image of the application, release it to Google Container Registry and deploy it on Google Cloud Run.
+
+<Tabs groupId="language">
+<TabItem value="Go">
+
+1. In the application directory, create a new Dagger module in a `ci` sub-directory.
+
+    ```shell
+    dagger init --name=my-module --sdk=go ci
+    cd ci
+    ```
+
+1. Install the Google Cloud Run module.
+
+    ```
+    dagger install github.com/vvaswani/daggerverse/google-cloud-run
+    ```
+
+1. Update the generated `dagger/main.go` file with the following code.
+
+    ```go file=./snippets/github-google-cloud/go/main.go
+    ```
+
+    This Dagger Module contains a constructor and three Dagger Functions. Let's look at each in detail.
+
+    The constructor accepts a single argument, a `Directory` representing the directory containing the application source code. This argument is available to all the Dagger Functions in the Dagger Module.
```

Comment:
To be more technically accurate, the argument is available to all the functions in the object (struct). Since this module only has one object, it remains true that it's available to all the functions in the module. And in cases where there's more objects and they also need this argument, you can still pass it around... but in that sense, so is any other argument that's received via a function, instead of the constructor.

Reply:
Modified